lib/libc/stdlib/lib_mktemp.c

```diff
@@ -71,26 +59,66 @@
  ****************************************************************************/
 
 /****************************************************************************
- * Public Functions
+ * Public Function Prototypes
  ****************************************************************************/
 
+char *__randname(char *, uint8_t);
+
 /****************************************************************************
  * Name: mktemp
  *
  * Description:
- *   The use of mktemp is dangerous; use mkstemp instead.
+ *   The mktemp() function generates a unique temporary filename from template.
+ *   The last six characters of template must be XXXXXX and these are replaced
+ *   with a  string  that  makes the filename unique.
+ *   Since it will be modified, template must not be a string constant,
+ *   but should be declared as a character array.
+ *
+ * Return Value:
+ *   The mktemp() function always returns template.
+ *   If a unique name was created,  the  last six  bytes  of template will have
+ *   been modified in such a way that the resulting name is unique
+ *   (i.e., does not exist already) If a unique name could not be created,
+ *   template is made an empty string, and errno is set to indicate the error.
  *
  ****************************************************************************/
 
-int mktemp(FAR char *path_template)
+FAR char *mktemp(FAR char *path_template)
 {
-	int fd = mkstemp(path_template);
-	if (fd < 0) {
-		return ERROR;
+	uint32_t retries = UINT8_MAX;
```

Comment:
In musl, retry value is set to 100. Is there any specific reason for increasing it to UINT8_MAX?
Also, why the variable is uint32_t ? why not uint8_t ? any specific reason for unsinged?

Reply:
1. I couldn't find the specific reason for magic number 100 and so I made it nearest max value UINT8_MAX. If you have any suggestion on this kindly let me know.
2. That variable type should be uint8_t.
3. As it is loop variable and the logic is to use 0 to UINT8_MAX.

Reply:
If we are not sure, lets keep the musl implementation as it is.